lib/debug/logging.py

```diff
@@ -0,0 +1,291 @@
+# Authors: see git history
+#
+# Copyright (c) 2010 Authors
+# Licensed under the GNU GPL version 3.0 or later.  See the file LICENSE for details.
+
+# basic info for inkstitch logging:
+# ---------------------------------
+# some idea can be found in: Modern Python logging - https://www.youtube.com/watch?v=9L77QExPmI0
+#
+# logging vs warnings (ChatGPT)
```

Comment:
ChatGPT?

Reply:
I couldn't find a better comparison than from GPT.

Reply:
Hmm, this is the first case where someone has contributed content generated by ChatGPT (that I know about).  Given the still murky licensing questions surrounding ChatGPT, I'd rather we did not include any content generated by it or other LLMs.  Could you please rewrite it in your own words?  It's okay if it's largely similar, I just want it not to be the text that was generated by ChatGPT.

Reply:
It hadn't occurred to me to consider it from that angle. By including the link to LLM, I just wanted to highlight that it's quite interesting from a mathematical perspective how such straightforward methods used by various AIs lead to such impressive results. Thank you very much for the heads up and explanation.